pkg/proxy/iptables/proxier.go

```diff
@@ -1609,11 +1618,29 @@ func (proxier *Proxier) syncProxyRules() {
 
 	// Finally, tail-call to the nodeports chain.  This needs to be after all
 	// other service portal rules.
-	writeLine(proxier.natRules,
-		"-A", string(kubeServicesChain),
-		"-m", "comment", "--comment", `"kubernetes service nodeports; NOTE: this must be the last rule in this chain"`,
-		"-m", "addrtype", "--dst-type", "LOCAL",
-		"-j", string(kubeNodePortsChain))
+	addresses, err := utilproxy.GetNodeAddresss(proxier.nodePortAddresses, proxier.networkInterface)
+	if err != nil {
+		glog.Errorf("Failed to get node ip address matching nodeport cidr")
+	} else {
+		args = append(args[:0],
+			"-A", string(kubeServicesChain),
+			"-m", "comment", "--comment", `"kubernetes service nodeports; NOTE: this must be the last rule in this chain"`,
+		)
+		addressList := addresses.UnsortedList()
+		if len(addressList) == 1 && utilproxy.IsZeroCIDR(addressList[0]) {
+			args = append(args,
+				"-m", "addrtype", "--dst-type", "LOCAL",
+				"-j", string(kubeNodePortsChain))
+			writeLine(proxier.natRules, args...)
+		} else {
+			// create nodeport rules for each IP one by one
+			iptablesDestIPListString := strings.Join(addressList, ",")
+			args = append(args,
+				"-d", iptablesDestIPListString,
```

Comment:
don't you still want the LOCAL check?  Otherwise this will capture cross-node nodeport accesses, I think...

Reply:
Sorry, I am not sure I fully understand...

Do you mean host A(1.2.3.4), host B(2.3.4.5)

From host A, access 2.3.4.5:port is not allowed? IIUC, it's allowed?

AFAIK, LOCAL check is intended to capture packets whose dest-address is "local". Am I missing something?